src/api/types.h

```diff
@@ -23,5 +23,27 @@ typedef struct CGameSettings {
   const char* seat_controllers[4];
   int num_controllers = 0;
 };
+
+typedef struct CObservedGameState {
+  int currentPlayer;
+  int turnNum;
+  int roundNum;
+  int riichiSticks;
+  int counters;
+  int lastCall;
+  int lastCaller;
+  bool concealedKan;
+  uint64_t seed;
+  CPiece pendingPiece;
+  int scores[4];
+  int points[4];  // Player points
+  bool hasRonned[4];
+  CPiece hands[4][14];
+  CPiece* players[4];  // Pointers to discard arrays for each player
```

Comment:
This is only 84 pieces at max ( 136 - 13x4 ), I think we should just make this inline in the object. That's not enough to be worth adding indirection, on linux the default stack size per thread is 8mb.